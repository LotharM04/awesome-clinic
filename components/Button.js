import React from "react";
import { Text, View, StyleSheet } from "react-native";

export default function Button(props) {
  const { title = "View", width, color } = props;
  return (
    <View
      style={[styles.button, { width: width, backgroundColor: color }]}
      data-testid={"button"}
    >
      <Text style={styles.text}>{title}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 10,
    paddingHorizontal: 30,
    borderRadius: 20,
    elevation: 3,
    marginTop: 20,
  },
  text: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: "bold",
    letterSpacing: 0.25,
    color: "white",
  },
});
