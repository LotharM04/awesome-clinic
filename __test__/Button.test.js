import React from "react";
import renderer from "react-test-renderer";
import Button from "../components/Button";

test("Button renders correctly", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
