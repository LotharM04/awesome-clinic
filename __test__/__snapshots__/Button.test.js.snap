// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  data-testid="button"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 3,
        "justifyContent": "center",
        "marginTop": 20,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      },
      Object {
        "backgroundColor": undefined,
        "width": undefined,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "letterSpacing": 0.25,
        "lineHeight": 21,
      }
    }
  >
    View
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  data-testid="button"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 3,
        "justifyContent": "center",
        "marginTop": 20,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      },
      Object {
        "backgroundColor": undefined,
        "width": undefined,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "letterSpacing": 0.25,
        "lineHeight": 21,
      }
    }
  >
    View
  </Text>
</View>
`;
